### DISCUSSION TITLE
Something similar to cache-first for Apollo and Urql?

### DISCUSSION THREAD

**Discussion by frederikhors** [TanStack/query#2128] [Q&A]
_I'm using the amazing [`svelte-query`](https://github.com/SvelteStack/svelte-query) and I opened this question [there too](https://github.com/SvelteStack/svelte-query/issues/38)._

Coming from Apollo or Urql I would like to know if there is a way to mimic (or improve) the behavior of policies like "`cache-first`":

> (the default) prefers cached results and falls back to sending an API request when no prior result is cached: https://formidable.com/open-source/urql/docs/basics/document-caching/#request-policies.

An example: I already have user data queried in Home page then I navigate to User page and I wanna say to svelte-query to NOT query again my user data because I already have them in cache (with Infinite stale).

Am I wrong to want this?

I tried with the `enabled` option but obviously if navigate on the user's page before the Home page the query does not start at all.

How to do?

**Comment by TkDodo** [comment]
`staleTime: Infinity` and `cacheTime: Infinity` should do exactly that. do you have an example that shows that this is not working please?
